### ISSUE TITLE
Add health check status to the dashboard

### ISSUE THREAD

**Issue by DamianEdwards** [microsoft/aspire#889]
The .NET Aspire ServiceDefaults project defaults to configuring health check endpoints for service projects. Similarly, .NET Aspire components default to registering health checks for the dependencies they integrate with. Currently, the status of these health checks is not visible anywhere by default. We should consider adding UI to the dashboard (add potentially [concepts to the AppHost app model](https://github.com/dotnet/aspire/issues/890)) to display the status of resources' health checks.

I can imagine a few click-stops:

1. Show last known status of health check endpoints, i.e. healthy, degraded, unhealthy, at a glance, perhaps on the relevant resources summary page, e.g. a column with a status indicator icon on the projects page. The interval the health check endpoints are inspected would default to something reasonable (e.g. 10 seconds) and perhaps be configurable either in the dashboard UI and/or in the AppHost app model.
2. Tracking of health check status changes over time, so that in addition to the last status, a history of status requests and their result could be viewed, e.g. clicking on the last known status indicator in the summary would open a view that shows the history of all status checks.
3. A detailed view of all the individual health checks that contribute to the overall status reported for a given health checks endpoint (similar to https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks#HealthCheckUI, perhaps we consider how we might just integrate that given that we're already depending on the individual health checks from this project from the relevant Aspire component packages)

**Comment by thompson-tomo** [comment]
@DamianEdwards I am thinking the same thing that it would be useful to have a location where a user can go to view the status of all their health checks as described in point 1.

For point 2 what i would like to see is: https://github.com/dotnet/aspnetcore/issues/53670 implemented so that the health check can be tracked over time using metrics. This approach would enable the re-utilization of the existing Metrics UI built for visualizing metrics (hence address point 3) and it benefit the whole eco-system as these health checks can then be utilized by any tool which implements OTEL metrics.

**Comment by alrz** [comment]
Could this possibly be enabled in the standalone dashboard?

**Comment by davidfowl** [comment]
@alrz Not likely since the standalone dashboard needs a resource server implementation. There isn't one today.

**Comment by adamint** [comment]
@drewnoakes can this be closed?

**Comment by DamianEdwards** [comment]
This is a different scenario, showing status of the health checks in the apps themselves, not the health checks that the app host is running on them.

**Comment by adamint** [comment]
@DamianEdwards right, we do show that in the dashboard UI right now. We can close this

**Comment by DamianEdwards** [comment]
@adamint the dashboard only knows about details of the health checks registered in the app host. If one manually adds an HTTP health check for a project then that will show up as just healthy or not, right? This still doesn't cover the scenario of displaying the full status of all health checks registered in a service project itself that contribute to the status returned from the HTTP health check. Or am I missing something?

**Comment by adamint** [comment]
Ah, I see now. Yes, you’re right

**Comment by joperezr** [comment]
As part of triage cleanup ahead of the 13.3 release, this issue is being moved to the Backlog milestone. If you believe it should remain in 13.3, feel free to re-add to the milestone and leave a comment here why it should not be punted.

**Comment by maddymontaquila** [comment]
Health check status is fully implemented in the Aspire Dashboard. Resources display their health status via `ResourceHealthStateValue`, the resource details panel shows an expanded health reports section (`HealthReports`), and the dashboard polls health check endpoints and displays pass/warn/fail/unknown states with icons and timestamps.

**Comment by davidfowl** [comment]
Reopening after a final strict review of the recent bulk closure. The associated change appears partial or does not fully satisfy the original issue, so this should remain open for normal triage.
